matsim/scenariogen/data/formats/eodmx.py

```diff
@@ -0,0 +1,423 @@
+
+
+def read_raw(household_file, person_file, trip_file, vivienda_file):
+    """ Read the input files into format used by conversion """
+
+    hh = pd.read_csv(household_file, encoding="utf-8", delimiter=",", decimal=".", low_memory=False, quoting=2)
+
+    p = pd.read_csv(person_file, encoding="utf-8", delimiter=",", decimal=".", low_memory=False, quoting=2)
+
+    t = pd.read_csv(trip_file, encoding="utf-8", delimiter=",", decimal=".", low_memory=False, quoting=2)
+
+    vv = pd.read_csv(vivienda_file, encoding="utf-8", delimiter=",", decimal=".", low_memory=False, quoting=2)
+
+    return hh, p, t, vv
+
+
+def convert(data: tuple, regio=None):
+    """ Convert eod data to standardized survey format """
+
+    (hh, pp, tt, vv) = data
+
+    ps = []
+    for p in pp.itertuples():
+        ps.append(
+            Person(
+                p_id=str(int(p.id_soc)),
+                p_weight=p.factor,
+                hh_id=str(int(p.id_hog)),
+                age=int(p.edad),
+                gender=EOD2017.gender(int(p.sexo)),
+                employment=EOD2017.employment(None if math.isnan(p.p3_7) else int(p.p3_7), int(p.edad)),
+                # no data on if person is able to be mobile or not, so always mobile
+                restricted_mobility=False,
+                driving_license=EOD2017.driv_lic_avail(p.edad),
+                car_avail=EOD2017.veh_avail(pint(hh[hh.id_hog == p.id_hog].p2_1_1)),
+                bike_avail=EOD2017.veh_avail(pint(hh[hh.id_hog == p.id_hog].p2_1_3)),
+                # pt abo always available as there are no abos in cdmx / zmvm
+                pt_abo_avail=Availability.YES,
+                # always true as mobile persons without trips will be filtered out anyway (later)
+                mobile_on_day= True if p.p4_2 == 1 else False,
+                # same  as mobile_on_day as there is no specific question on this
+                present_on_day= True if p.p4_2 == 1 else False,
+                # there is no data on which weekday it is, so we just assume it always is a tuesday = valid
+                reporting_day=2,
+                n_trips=0 if math.isnan(p.p5_4) else pint(p.p5_4),
+                home_district=str(int(p.distrito)).zfill(3)
+            )
+        )
+
+    ps = pd.DataFrame(ps).set_index("p_id")
+
+    hhs = []
+    for h in hh.itertuples():
+
+        hh_id = str(pint(h.id_hog))
+        hhs.append(
+            Household(
+                hh_id=hh_id,
+                h_weight=h.factor,
+                n_persons=pint(vv[vv.id_viv == h.id_viv].p1_1),
+                n_cars=pint(h.p2_1_1),
+                n_bikes=pint(h.p2_1_3),
+                n_other_vehicles=pint(h.p2_1_2),
+                # no data on parking position -> all NA
+                car_parking=ParkingPosition.NA,
+                economic_status=EOD2017.economic_status(pint(h.estrato)),
+                # no info on hh type given in data
+                type=HouseholdType.UNKNOWN,
```

Comment:
Are all persons of the household available? In this case, you could compute this value. But it might not be super relevant, I think it is not used anywhere at the moment.

Reply:
No, this is a survey, where random _people_ are selected to represent their household. Therefore, not all persons of the household are available. I think it does not matter, as the parameter is not used for activity sampling anyways.